src/parser.pegjs

```diff
@@ -46,16 +46,42 @@
 // ---- Syntactic Grammar -----
 
 Grammar
-  = __ topLevelInitializer:(@TopLevelInitializer __)? initializer:(@Initializer __)? rules:(@Rule __)+ {
+  = __ imports:(@Import __)* topLevelInitializer:(@TopLevelInitializer __)? initializer:(@Initializer __)? rules:(@Rule __)+ {
       return {
         type: "grammar",
+        imports,
         topLevelInitializer,
         initializer,
         rules,
         location: location()
       };
     }
 
+Import
+  = 'import' __
+    '{' __ rules:ImportedRule|.., __ ',' __| (__ ',')? __ '}' __
```

Comment:
Imagine a situation where you have a grammar that defines an `expression` rule (for example, this exact grammar) that imports another grammar with `expression` rule (ecmascript grammar for actions).

I think it would be best just to reuse the whole `import` rule from JS, and emit it verbatim into generated file.

Reply:
> I think it would be best just to reuse the whole import rule from JS, and emit it verbatim into generated file.

I deliberately chose to support only that subset of JS `import` clause where the rule names are explicitly indicated. This will allow us to make basic checks (e.g. that the rules we refer to exist, or that we do not have two rules with the same name from different imports -- you will need to rename one of them using `as`)

Reply:
I assume that all rules are exported from every .peggy file?  There's no way to mark something `private`, or to explicitly export?

Reply:
That is the question, that should be investigated; yes, without annotations we have no way to mark what we should export, and what not (at least, _inside_ the grammar).